src/main/java/uk/gov/ons/census/fwmt/jobservice/service/routing/nc/NcCeCreateEnglandAndWales.java

```diff
@@ -67,49 +66,45 @@ public boolean isValid(FwmtActionInstruction rmRequest, GatewayCache cache) {
   @Override
   public void process(FwmtActionInstruction rmRequest, GatewayCache cache, Instant messageReceivedTime)
       throws GatewayException {
-    GatewayCache previousDetails = cacheService.getById(rmRequest.getOldCaseId());
-    String newCaseId = String.valueOf(UUID.randomUUID());
-    String accessInfo = null;
-    String careCodes = null;
+    String ncCaseId = rmRequest.getCaseId();
+    String originalCaseId = rmRequest.getOldCaseId();
+    GatewayCache originalCache = cacheService.getById(originalCaseId);
+    if (originalCache == null) {
+      String reason = "Original case does not exist with cache";
+      String NC_CSV_LOAD_FAILURE = "NC_CSV_LOAD_FAILURE";
```

Comment:
externalise is? If we don't already have one, please create a class that contains these. - I think there is one part of the events library, don't bother updating that one, just have one specific to this service? I'll be triggering different error events soon, so will be good to have them in one place

Reply:
will do